libgnucash/app-utils/gnc-option-impl.hpp

```diff
@@ -0,0 +1,1066 @@
+        else
+            throw std::invalid_argument("Validation failed, value not set.");
+    }
+    void set_default_value(ValueType value)
+    {
+        if (this->validate(value))
+            m_value = m_default_value = value;
+        else
+            throw std::invalid_argument("Validation failed, value not set.");
+    }
+    void get_limits(ValueType& upper, ValueType& lower, ValueType& step) const noexcept
+    {
+        upper = m_max;
+        lower = m_min;
+        step = m_step;
+    }
+    void reset_default_value() { m_value = m_default_value; }
+    bool is_changed() const noexcept { return m_value != m_default_value; }
+    GncOptionUIType get_ui_type() const noexcept { return m_ui_type; }
+    void make_internal() { m_ui_type = GncOptionUIType::INTERNAL; }
+    bool is_alternate() const noexcept { return m_alternate; }
+    void set_alternate(bool value) noexcept {
+        if (m_ui_type == GncOptionUIType::PLOT_SIZE)
+            m_alternate = value;
+    }
+private:
+    GncOptionUIType m_ui_type = GncOptionUIType::NUMBER_RANGE;
+    ValueType m_value;
+    ValueType m_default_value;
+    ValueType m_min;
+    ValueType m_max;
+    ValueType m_step;
+    bool m_alternate = false;
+};
+
+template<class OptType,
+         typename std::enable_if_t<is_RangeValue_v<OptType>, int> = 0>
+inline std::ostream&
+operator<< (std::ostream& oss, const OptType& opt)
+{
+    if (opt.get_ui_type() == GncOptionUIType::PLOT_SIZE)
+        oss << (opt.is_alternate() ? "pixels" : "percent") << " ";
+    oss << opt.get_value();
+    return oss;
+}
+
+template<class OptType,
+         typename std::enable_if_t<is_RangeValue_v<OptType>, int> = 0>
+inline std::istream&
+operator>> (std::istream& iss, OptType& opt)
+{
+    if (opt.get_ui_type() == GncOptionUIType::PLOT_SIZE)
+    {
+        std::string alt;
+        iss >> alt;
+        opt.set_alternate(strncmp(alt.c_str(), "percent",
+                                  strlen("percent")) == 0);
+    }
+    if constexpr (std::is_same_v<std::decay_t<OptType>,
+                  GncOptionRangeValue<double>>)
+    {
+        double d;
+        iss >> d;
+        opt.set_value(d);
+    }
+    else
+    {
+        int i;
+        iss >> i;
+        opt.set_value(i);
+    }
+    return iss;
+}
+
+using GncMultichoiceOptionEntry = std::tuple<const std::string,
+                                             const std::string,
+                                             GncOptionMultichoiceKeyType>;
+using GncMultichoiceOptionIndexVec = std::vector<std::size_t>;
+using GncMultichoiceOptionChoices = std::vector<GncMultichoiceOptionEntry>;
+
+/** @class GncOptionMultichoiceValue
+ * Multichoice options have a vector of valid options
+ * (GncMultichoiceOptionChoices) and validate the selection as being one of
+ * those values. The value is the index of the selected item in the vector. The
+ * tuple contains three strings, a key, and a display
+ * name, which * should be localized at the point of use.
```

Comment:
Something went wrong while composing this comment. Only two strings are listed.

Reply:
Another left-over from removing the tooltips on choices: The third string was the tooltip string.